Feat: 문제집 생성

adorableco opened the pull request:
## **PR**

### ✨ 작업 내용

- 문제집 생성 기능을 구현함


### ✨ 참고 사항

- 현재는 `authentication.name` 에 member email이 담겨 있어서 findByEmail로 문제집 생성자를 할당하고 있는데, `authentication.name` 에 memberId (UUID) 가 담기는게 더 낫지 않을까 고민입니당
-  @RinRinPARK 근데 어제 말한 UserResolver ? 를 사용하면 거기서 사용자 정보도 처리할 수 있는거려나??
- 현재 문제집 생성 성공 응답에 생성된 문제 id만 담는데 더 추가할게 있으면 알려주세요


### ✅ 응답 예시


<img width="499" alt="스크린샷 2024-07-08 오후 12 43 16" src="https://github.com/swm-standard/Phote_BE/assets/90397541/8c1c6152-696d-4caa-972a-9ee963b6143e">


---

### ⏰ 현재 버그

x

---

### ✏ Git Close #7

adorableco committed:
Feat: 문제집 생성

part of SWM-36

adorableco committed:
Fix: build.gradle.kts 의존성 중복 삭제

part of SWM-36

adorableco committed:
Fix: authentication 헤더 이름 변경

- `Authentication` -> `accessToken`

part of SWM-36

RinRinPARK commented on src/main/kotlin/com/swm_standard/phote/dto/WorkbookDtos.kt:
```diff
@@ -0,0 +1,22 @@
+data class WorkbookCreationRequest(
+    @field:NotBlank
+    @JsonProperty("title")
+    private val _title: String?,
```
지금 title이 request로 안들어오면 500이 뜨는데 커스텀에러로 400대 뜨게 고쳐주면 좋을 것 같아여

RinRinPARK reviewed (changes requested):
```- 현재는 authentication.name 에 member email이 담겨 있어서 findByEmail로 문제집 생성자를 할당하고 있는데, authentication.name 에 memberId (UUID) 가 담기는게 더 낫지 않을까 고민입니당```
-> 저도 글케 생각합니다~~ uuid가 나을 듯

```@RinRinPARK 근데 어제 말한 UserResolver ? 를 사용하면 거기서 사용자 정보도 처리할 수 있는거려나??```
-> accessToken받아서 그걸로 자동으로 UUID로 변환해서 request에 뿌려줄 생각입니당. 그럼 uuid가지고 사용자 정보 받아서 쓰면 될 듯요??

```현재 문제집 생성 성공 응답에 생성된 문제 id만 담는데 더 추가할게 있으면 알려주세요 ```
--> 충분하다고 생각합니돠

- **지금 문제집 생성하면 디비에 emoji가 그냥 null로 들어가있는데, 자동으로 이모지 채워주는 부분 코드 추가 작성 필요할 듯합니다**
- **명세서에 request, response 수정이 필요해보입니다**

RinRinPARK commented on src/main/kotlin/com/swm_standard/phote/dto/WorkbookDtos.kt:
```diff
@@ -0,0 +1,22 @@
+import jakarta.validation.constraints.NotBlank
+import java.util.UUID
+
+data class WorkbookCreationRequest(
```
아 그리고 이부분 우리 코드컨벤션을 맞추는게 좋을 것 같은데, 개인적으로 createWorkbookRequest/createWorkbookResponse 이런 식으로 기능명이랑 dto 이름 맞추는게 좋을 것 같은데 어떠세여??

adorableco replied:
좋아염~~

adorableco committed:
Fix: 문제집 생성 수정

- `title`이 blank 인 경우 예외 처리를 4xx 번으로 수정
- emoji 기본값 추가
- dto 이름 통일

part of SWM-36
related to #7
